packages/arb-token-bridge-ui/src/hooks/useNetworksAndSigners.tsx

```diff
@@ -258,7 +260,11 @@ export function NetworksAndSignersProvider(
               provider: l2Provider
             },
             isConnectedToArbitrum: false,
-            chainId: l1Network.chainID
+            chainId: l1Network.chainID,
+            isSmartContractWallet: await addressIsSmartContract(
```

Comment:
This seems to work, but just sanity checking the idea:

We only need to check if the address is a SC on the connected chain. 

If e.g., address on connected chain A has no bytecode but is in fact a SC on chain B, we don't care / don't need to label it a "smart contract wallet" because in this case it can't sign transactions on chain A anyway. Ya?

Reply:
yes, if we are connected to L1 we check bytecode on L1 provider, if connected to L2 we check bytecode on L2 provider